### DISCUSSION TITLE
How can I run gradle-groovy project in local ?

### DISCUSSION THREAD

**Discussion by yonginscv** [naver/ngrinder#782] [Q&A] [answered]
Hello, 

Could you please let me know how can I run ngrinder's gradle-groovy project in local ( IntelliJ or VSCode ) ?

I used default project ngrinder-controller 3.5.3 gernerate ( build.gradle, src/main/java/TestRunner.groovy )
but cant figure out how to run TestRunner.goovy

Thank you.

**Comment by donggyu04** [comment, marked as answer]
@yonginscv 

Hi, please follow the steps below. I will explain based on the IntelliJ IDE.

1. Checkout your scripts from the repository that commonly svn like `svn checkout http://localhost:8080/svn/admin/groovy-gradle-exam`

![image](https://user-images.githubusercontent.com/14273601/114416103-4dcf8e80-9beb-11eb-8c35-299df3b61683.png)

2. Import the project using the Intellij IDE.
3. Open the Preference(`command + ,` on macOS) and change the build and run option using Intellij IDEA, not Gradle.

![image](https://user-images.githubusercontent.com/14273601/114416672-e108c400-9beb-11eb-93b0-68fb0bb4c343.png)

4. Right-click the `TestRunner.groovy` and click `Run TestRunner`
> If your `ngrinder-groovy` scope is `compileOnly`, change it to `implementation` like below

```
dependencies {
    // compileOnly (group: "org.ngrinder", name: "ngrinder-groovy", version: ngrinder_version)
    implementation (group: "org.ngrinder", name: "ngrinder-groovy", version: ngrinder_version)
}
```

5. You will probably see the below message in the console with errors.

![image](https://user-images.githubusercontent.com/14273601/114417600-b1a68700-9bec-11eb-8c0b-f02877697c09.png)

6. Copy the `-javaagent:/...` option and paste to VM options in Run/Debug configurations panel.
<img src="https://user-images.githubusercontent.com/14273601/114418043-19f56880-9bed-11eb-8e4c-e0803d439cf5.png" width="650" height="220" />

7. Run your test again.

![image](https://user-images.githubusercontent.com/14273601/114418471-7f495980-9bed-11eb-9b1a-6485d9a53ca2.png)
